### ISSUE TITLE
Boot menu with XBian versions

### ISSUE THREAD

**Issue by CurlyMoo** [xbianonpi/xbian-package-initramfs#7]
I just had a nice idea. Since we are running BTRFS, we can easily create snapshots before each version update (or other package updates). However, not all users report all our updates as an improvement. So, what about creating a boot menu with which we enable users to boot from one of the snapshots created of prior versions? This way, users can always boot from a previous version without reinstalling, and easily update to the next version without reinstalling the latest version again. He just has to boot the last version and update (which again adds another boot menu item). This boot menu can show for about 3 sec. so it doesn't slow down the boot too much.

Something for version 2 i think :smile:

**Comment by CurlyMoo** [comment]
@mk01, I was thinking about our talk on IRC yesterday and that hardly anyone understands how our failsafe mechanisms work. apt downgrade, btrfs snapshots rollback / boot. I think a key feature to make users aware is to add this bootmenu with the 3 snapshots. Then users always `see` that XBian has snapshots and that they can do things with it.

I also think we should add a message in xbian-config and in the shell that explains that users can always downgrade when apt upgrade fails. In xbian-config, it should be made easy to actually downgrade.

**Comment by mk01** [comment]
@CurlyMoo 

better idea - let's finally do "tips". showing at console upon login. oneliners. in the meaning:
```
Did you know, that .... ?
```

**Comment by CurlyMoo** [comment]
Yes, but only users using the command line will see those tips. Not regular users.

**Comment by mk01** [comment]
@belese

**Comment by belese** [comment]
Ahh, i've done code fr that few times ago, but wasn't able to do it working, 
the menu was working,  a file is writed on boot partition for action on next boot
but don't know after.
will look if i find code, but i'm believe it is lost, and i probably don't have a backup, have to check at home
I was using the splash screen with menu item, and time out of each selected items, once you want to select, you reboot.

But i'm now tring to add bluetooth (and A2dp sink) directly inside xbmc, and it takes a bit times,  i've already extends DBus messaging, add a DBus handler , hope to have a first test soon (or not;-), 
it should normally work with bluez4 and 5.
After that, i could look again in that, but no prmoise when i could do it.
